libarchive/test/test_write_disk_perms.c

```diff
@@ -201,6 +201,49 @@ DEFINE_TEST(test_write_disk_perms)
 	failure("dir_overwrite_0744: st.st_mode=%o", st.st_mode);
 	assertEqualInt(st.st_mode & 0777, 0744);
 
+	/* For dir, the owner should get left when not overwritting. */
+	assertMakeDir("dir_owner", 0744);
+	if (getuid() == 0) {
```

Comment:
I think you can simplify this test code.  Both branches expect the UID to be unchanged, so I think you can:
```
if (getuid() == 0) {
  original_uid = getuid() + 1;
  try_to_change_uid = getuid();
  assertEqualInt(0, chown("dir_owner", original_uid));
} else {
  original_uid = getuid();
  try_to_change_uid = getuid() + 1;
}
/*  ... verify original_uid, try to restore to try_to_change_uid, then verify original_uid again */
```

Reply:
Good idea, I'll look to making that change.